WooCommerce/Classes/Tools/UnitInputFormatter/IntegerInputFormatter.swift

```diff
@@ -11,18 +16,28 @@ struct IntegerInputFormatter: UnitInputFormatter {
 
     func isValid(input: String) -> Bool {
         guard input.isEmpty == false else {
-            // Allows empty input to be replaced by 0.
+            // Allows empty input to be replaced by defaultValue
             return true
         }
-        return numberFormatter.number(from: input) != nil
+        return numberFormatter.number(from: input) != nil || input == "-"
     }
 
     func format(input text: String?) -> String {
         guard let text = text, text.isEmpty == false else {
-            return "0"
+            return defaultValue
         }
 
-        let formattedText = numberFormatter.number(from: text)?.stringValue ?? "0"
+        var formattedText = numberFormatter.number(from: text)?.stringValue ?? defaultValue
+
+        // The minus sign is mantained if present
+        let minus: Character = "-"
```

Comment:
nit: how about having this as a private constant so that `isValid(input:)` and L35 can also use this?